fpm/src/fpm.f90

```diff
@@ -114,21 +131,111 @@ subroutine cmd_build(settings)
 subroutine cmd_install()
     print *, "fpm error: 'fpm install' not implemented."
     error stop 1
-end subroutine
-
-subroutine cmd_new()
-    print *, "fpm error: 'fpm new' not implemented."
-    error stop 1
-end subroutine
+end subroutine cmd_install
+
+subroutine cmd_new() ! --with-executable F --with-test F '
+character(len=:),allocatable :: message(:)
+logical                      :: cli_with_executable  ! command line keyword value set by get_args(3f)
+logical                      :: cli_with_test        ! command line keyword value set by get_args(3f)
+character(len=:),allocatable :: dirname, basename
+   call get_args('with-executable',cli_with_executable)                         ! get command line arguments
+   call get_args('with-test',cli_with_test)
+   ! assume everything unclaimed by keywords on the command line are command arguments for new command
+   if(size(words).eq.2.and.len(words).gt.0)then
+      dirname=trim(words(2))
+      basename=dirname !! basename(dirname)
+      !! are full pathnames allowed? Is more than one pathname allowed?
+   elseif(size(words).gt.2)then
+     write(stderr,'(a)') 'fpm::new<ERROR> more than one directory name'
+     write(stderr,'(a)') '      usage: fpm new DIRECTORY_NAME --with-executable --with-test'
+     stop 3
+   else                                                                     ! no directory name to create or update on commandline
+     write(stderr,'(a)') 'fpm::new<ERROR> missing directory name'
+     write(stderr,'(a)') '      usage: fpm new DIRECTORY_NAME --with-executable --with-test'
+     stop 4
+   endif
+
+   message=[character(len=80) :: &                                           ! create fpm.toml
```

Comment:
```suggestion
   message=[&                                           ! create fpm.toml
```

I don't think you need to specify `character(len=...)` here - it's actually a little misleading with arbitrary length `80`. And similarly below.